core/src/env_platform/env_service.bal

```diff
@@ -82,6 +99,45 @@ service envservice on ep0 {
     }
     resource function putApplicationById(http:Caller caller, http:Request req, string applicationId, TreeRemovalForm body) returns error? {
 
+        http:Response response = new;
+        string authHeader = req.getHeader("Authorization");
+        [string, string]|error userInfoFromJWT = getUserInfoFromJWT(authHeader);
+        if (userInfoFromJWT is [string, string]) {
+            [string, string] [userId, userType] = userInfoFromJWT;
+            boolean|error applicationBelongsToUserResult = applicationBelongsToUser(applicationId, userId);
+            if (applicationBelongsToUserResult is error) {
+                response.statusCode = http:STATUS_NOT_FOUND;
+                if (applicationBelongsToUserResult.reason() == "No applications") {
+                    response.setPayload({"reason": "No applications for given user."});
```

Comment:
I think we should see if we can re-use some of the reasons that we append to errors we return from our own functions here rather than just abandoning the values altogether. However, this needs to be done carefully and **only** where needed.

Reply:
I mentioned this on my review @yashodgayashan what happened to this?

Reply:
Will update those functions by tomorrow.

Reply:
Will resolved with the latest PR